src/D2L.CodeStyle.Analyzers/CustomerState/CustomerStateAnalyzer.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using D2L.CodeStyle.Analyzers.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace D2L.CodeStyle.Analyzers.CustomerState {
+
+	[DiagnosticAnalyzer( LanguageNames.CSharp )]
+	public sealed class CustomerStateAnalyzer : DiagnosticAnalyzer {
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(
+			Diagnostics.SingletonDependencyHasCustomerState
+		);
+
+		public override void Initialize(
+			AnalysisContext context
+		) {
+			//context.EnableConcurrentExecution();

```

Comment:
Why commented out?

Reply:
Whoops.  Let get the caching you wanted in, and whatever you wanted to talk about and we'll take another swing at this review.